cpp/src/arrow/compute/kernels/hash.h

```diff
@@ -66,6 +47,29 @@ Status GetDictionaryEncodeKernel(FunctionContext* ctx,
 ARROW_EXPORT
 Status Unique(FunctionContext* context, const Datum& datum, std::shared_ptr<Array>* out);
 
+// Constants for accessing the output of ValueCounts
+ARROW_EXPORT extern const char kValuesFieldName[];
+ARROW_EXPORT extern const char kCountsFieldName[];
+ARROW_EXPORT extern const int32_t kValuesFieldIndex;
+ARROW_EXPORT extern const int32_t kCountsFieldIndex;
```

Comment:
Hm, I'm having a YAGNI reaction to this. Premature development?

Reply:
For all of them?  I think we should probably provide one or another so there isn't ambiguity?